apps/tx-builder/src/components/Dashboard.tsx

```diff
@@ -92,74 +70,141 @@ const Dashboard = (): ReactElement => {
-          to={address}
+      <AddNewTransactionFormWrapper>
+        <StyledTitle size="sm">Multisend transaction builder</StyledTitle>
+
+        <StyledText size="sm">
+          This app allows you to build a custom multisend transaction. Enter a Ethereum contract address or ABI to get
+          started.{' '}
+          <Link
+            href="https://help.gnosis-safe.io/en/articles/4680071-create-a-batched-transaction-with-the-transaction-builder-safe-app"
+            size="lg"
+            target="_blank"
+            rel="noreferrer"
+          >
+            Learn how to use the transaction builder.
+          </Link>
+        </StyledText>
+
+        {/* Address Input */}
+        <StyledAddressInput
+          id="address"
+          name="address"
+          label="Enter Address or ENS Name"
+          hiddenLabel={false}
+          address={address}
+          showNetworkPrefix={!!chainInfo?.shortName}
           networkPrefix={chainInfo?.shortName}
+          error={isAddressInputFieldValid ? '' : 'The address is not valid'}
+          showLoadingSpinner={isABILoading}
+          showErrorsInTheLabel={false}
           getAddressFromDomain={getAddressFromDomain}
-          nativeCurrencySymbol={chainInfo?.nativeCurrency.symbol}
-          onRemoveTransaction={handleRemoveTransaction}
-          onSubmitTransactions={handleSubmitTransactions}
+          onChangeAddress={(address: string) => setAddress(address)}
+          InputProps={{
+            endAdornment: contractHasMethods && isValidAddress(address) && (
+              <InputAdornment position="end">
+                <CheckIconAddressAdornment />
+              </InputAdornment>
+            ),
+          }}
         />
-      )}
+
+        {/* ABI Warning */}
+        {loadContractError && (
+          <StyledWarningText color="warning" size="lg">
+            No ABI found for this address
+          </StyledWarningText>
+        )}
+
+        <JsonField id={'abi'} name="abi" label="Enter ABI" value={abi} onChange={setAbi} />
+
+        {showNewTransactionForm && (
+          <AddNewTransactionForm
+            onAddTransaction={handleAddTransaction}
+            contract={contract}
+            to={address}
+            networkPrefix={chainInfo?.shortName}
+            getAddressFromDomain={getAddressFromDomain}
+            nativeCurrencySymbol={chainInfo?.nativeCurrency.symbol}
+          />
+        )}
+      </AddNewTransactionFormWrapper>
+
+      {/* Transactions Batch section */}
+      <TransactionsSectionWrapper>
+        {transactions.length > 0 ? (
+          <TransactionsBatchList
+            transactions={transactions}
+            onRemoveTransaction={handleRemoveTransaction}
+            onSubmitTransactions={handleSubmitTransactions}
+            handleRemoveAllTransactions={handleRemoveAllTransactions}
+            handleReorderTransactions={handleReorderTransactions}
+            showTransactionDetails={false}
+            allowTransactionReordering
```

Comment:
you can use `showTransactionDetails` & `allowTransactionReordering` to configure the `TransactionsBatchList` component:

https://user-images.githubusercontent.com/26763673/154505211-ec7f41ae-e987-4ff8-8690-5f9f75ef4d6e.mov

Reply:
I am working in [this other ticket](https://github.com/gnosis/safe-react-apps/issues/320) about review and confirm screen.
branch: `feature/tx-builder-review-and-confirm-screen`